src/recipes/resource-saver-loader.md

```diff
@@ -37,11 +37,17 @@ unsafe impl ExtensionLibrary for MyGDExtension {
         }
     }
 
-    // Unregisters the singleton when the extension exits.
-    fn on_level_init(level: InitLevel) {
+    // Unregister the singleton when the extension exits.
+    fn on_level_deinit(level: InitLevel) {
         if level == InitLevel::Scene {
-            Engine::singleton().unregister_singleton("MyAssetSingleton");
-            my_singleton.free();
+            let mut engine = Engine::singleton();
+            let singleton_name = "MyAssetSingleton";
+            if let Some(my_singleton) = engine.get_singleton(singleton_name) {
+                engine.unregister_singleton(singleton_name);
+                my_singleton.free();
+            } else {
+                godot_error!("Failed to get singleton");
+            }
```

Comment:
When exactly can this fail? After all, we registered it during init.

Generally I'd rather not encourage defensive programming if it's not necessary, it creates uncertainty about the reliability of operations. If there's a valid reason why this may fail, it should be explicitly documented.

Reply:
Let's not copy-paste bad habits but fix them at the root 🙂

For the sake of this example, which is about resource savers/loaders, we don't need to be concerned about someone prematurely unregistering the singleton. Let's focus on the core logic to demonstrate 😉 ideally we'll provide a more automated way of registering singletons at some point anyway.